### PR TITLE
Navigation

### PR DESCRIPTION
**Pull request by EngOmarElsayed**
### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/swiftui/concepts/gestures/TapgestuerOutput.png
Comment (SSwiniarski) [inline content/swiftui/concepts/gestures/TapgestuerOutput.png:1]: It looks like you branched off of a prior PR rather than main, which means you have extra extraneous files in this PR. This one needs to be removed.

File: content/swiftui/concepts/gestures/gestures.md
Comment (SSwiniarski) [inline content/swiftui/concepts/gestures/gestures.md:1]: It looks like you branched off of a prior PR rather than main, which means you have extra extraneous files in this PR. This one needs to be removed.

File: content/swiftui/concepts/navigation/navigation.md
Comment (SSwiniarski) [inline content/swiftui/concepts/navigation/navigation.md:3]:
```suggestion
Description: 'Navigation is the ability to navigate to different views from the root view in the app.'
```

File: content/swiftui/concepts/navigation/navigation.md
Comment (SSwiniarski) [inline content/swiftui/concepts/navigation/navigation.md:16]:
```suggestion
**Navigation** is the ability to navigate to different views from the root view in the app. In SwiftUI, `NavigationStack` is used to enable the root view to present views over it.  In iOS 15 and before, `NavigationView` was used, but it's been deprecated since iOS 16 introduced `NavigationStack`. 
```

File: content/swiftui/concepts/navigation/navigation.md
Comment (SSwiniarski) [inline content/swiftui/concepts/navigation/navigation.md:19]:
```suggestion
## NavigationStack

To be able to navigate to different views from the root view, the view will be warped with `NavigationStack` struct as in the example below: 
```

File: content/swiftui/concepts/navigation/navigation.md
Comment (SSwiniarski) [inline content/swiftui/concepts/navigation/navigation.md:30]:
```suggestion

This is the first step to be able to navigate to different views. This defines the root view of the stack. Then to define the destination use `NavigationLink`. When pressed, it navigates to the destination view that was defined: 
```

File: content/swiftui/concepts/navigation/navigation.md
Comment (SSwiniarski) [inline content/swiftui/concepts/navigation/navigation.md:50]:
```suggestion

This code will output:

![This is a gif that shows Navigation in action](https://raw.githubusercontent.com/Codecademy/docs/main/media/NavigationStack.gif)
```

File: content/swiftui/concepts/navigation/navigation.md
Comment (SSwiniarski) [inline content/swiftui/concepts/navigation/navigation.md:52]:
```suggestion
The label of the `NavigationLink` can be customized:
```
We avoid first and second-person pronouns in Docs.

File: content/swiftui/concepts/navigation/navigation.md
Comment (SSwiniarski) [inline content/swiftui/concepts/navigation/navigation.md:75]:
```suggestion

This code will output:

![This is a gif that shows Navigation with a custom label in action](https://raw.githubusercontent.com/Codecademy/docs/main/media/NavigationStack-with-Custom-label.gif)
```

File: content/swiftui/concepts/navigation/navigation.md
Comment (SSwiniarski) [inline content/swiftui/concepts/navigation/navigation.md:80]:
```suggestion
## NavigationStack with Lists

One of the common examples of using `NavigationStack` is with `Lists`. For example, there is a list of names in the view. The goal is to be able to navigate to another view by pressing the element in the list. This can be achieved by wrapping the `List` in a `NavigationStack` :
```

File: content/swiftui/concepts/navigation/navigation.md
Comment (SSwiniarski) [inline content/swiftui/concepts/navigation/navigation.md:100]:
```suggestion

This code will output:

![This is a gif that shows how to use NavigationStack with Lists](https://raw.githubusercontent.com/Codecademy/docs/main/media/NavigationStack-with-List.gif)
```

File: content/swiftui/concepts/navigation/navigation.md
Comment (SSwiniarski) [inline content/swiftui/concepts/navigation/navigation.md:105]:
```suggestion
## NavigationStack Modifiers

There are some basic modifiers to use with `NavigationStack`. The first one is `.navigationTitle`. This is used to be able to declare the title of the view. It looks like this:
```

File: content/swiftui/concepts/navigation/navigation.md
Comment (SSwiniarski) [inline content/swiftui/concepts/navigation/navigation.md:125]:
```suggestion

This code will output:

![This picture shows the effect of using NavigationStack with navigationTitle](https://raw.githubusercontent.com/Codecademy/docs/main/media/NavigationStack-with-navigationTitle.png)
```

File: content/swiftui/concepts/navigation/navigation.md
Comment (SSwiniarski) [inline content/swiftui/concepts/navigation/navigation.md:127]:
```suggestion
Another important modifier is `.navigationBarBackButtonHidden()`. This modifier removes the back button that appears when navigating to another view. This modifier is applied to the desired view:
```

File: content/swiftui/concepts/navigation/navigation.md
Comment (SSwiniarski) [inline content/swiftui/concepts/navigation/navigation.md:148]:
```suggestion

This code produces the following output: 

![This gif shows the effect of using NavigationStack with navigationBarBackButtonHidden](https://raw.githubusercontent.com/Codecademy/docs/main/media/NavigationStack-with-navigationBarBackButtonHidden.gif)
```

File: content/swiftui/concepts/navigation/navigation.md
Comment (SSwiniarski) [inline content/swiftui/concepts/navigation/navigation.md:152]:
```suggestion
The modifiers above are just two examples out of many.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@EngOmarElsayed I did a first review and left a few comments. One thing to note is because you forked from a prior PR, there are a couple of files in this one that need to be removed. This should not affect your prior branch.

File: content/swiftui/concepts/gestures/TapgestuerOutput.png
Comment (EngOmarElsayed) [reply content/swiftui/concepts/gestures/TapgestuerOutput.png:1]: Sorry 😅

**Review by SSwiniarski — APPROVED** [review]
@EngOmarElsayed I made one small edit and it looks good for a second review.

**Comment by EngOmarElsayed** [comment]
Hey 👋 it's been 2 weeks and no updates, just checking. because I want to start on a new doc. I know you are busy I don't mean to put you under pressure, I am just checking

**Comment by SSwiniarski** [comment]
@EngOmarElsayed have patience, another maintainer will review this. There's just a bit of a backlog. In the meantime, there's no reason you can't work on something else. Just make sure you create the new branch for it off of your main fork, and not this branch.

File: content/swiftui/concepts/navigation/navigation.md
Comment (yangc95) [inline content/swiftui/concepts/navigation/navigation.md:18]:
```suggestion
## `NavigationStack`
```

File: content/swiftui/concepts/navigation/navigation.md
Comment (yangc95) [inline content/swiftui/concepts/navigation/navigation.md:20]:
```suggestion
To be able to navigate to different views from the root view, views will be wrapped with a `NavigationStack` `struct` as in the example below: 
```

File: content/swiftui/concepts/navigation/navigation.md
Comment (yangc95) [inline content/swiftui/concepts/navigation/navigation.md:22]:
```suggestion
```swift
```

Comment (yangc95) [reply content/swiftui/concepts/navigation/navigation.md:22]: These should all be lowercased ^^

File: content/swiftui/concepts/navigation/navigation.md
Comment (yangc95) [inline content/swiftui/concepts/navigation/navigation.md:85]:
```suggestion
## `NavigationStack` with Lists
```

Comment (yangc95) [reply content/swiftui/concepts/navigation/navigation.md:85]: Style: Wrap in backticks even when in title

File: content/swiftui/concepts/navigation/navigation.md
Comment (yangc95) [inline content/swiftui/concepts/navigation/navigation.md:162]: For all these gifs, can you crop out the safe areas?

File: content/swiftui/concepts/navigation/navigation.md
Comment (yangc95) [inline content/swiftui/concepts/navigation/navigation.md:32]: Be a bit definitive and less directional in language. So rather than "This is the first step" -> "To navigate to different views..."

File: content/swiftui/concepts/navigation/navigation.md
Comment (EngOmarElsayed) [reply content/swiftui/concepts/navigation/navigation.md:164]: Nop I can't

Comment (yangc95) [reply content/swiftui/concepts/navigation/navigation.md:164]: Do you need assistance with this?

File: content/swiftui/concepts/navigation/navigation.md
Comment (yangc95) [inline content/swiftui/concepts/navigation/navigation.md:25]:
```suggestion
@main
struct MyApp: View {
```

File: content/swiftui/concepts/navigation/navigation.md
Comment (EngOmarElsayed) [reply content/swiftui/concepts/navigation/navigation.md:163]: yes please I don’t know how to do that, because this gif is generated from the sim. I don’t have the capability to edit them. @yangc95

**Comment by EngOmarElsayed** [comment]
is there anything thing left I need to do.

**Review by yangc95 — APPROVED** [review]

**Comment by yangc95** [comment]
> is there anything thing left I need to do.

@EngOmarElsayed you're good! This is ready for merge now 🙂

**Pull request merged**
